atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/LockStore.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.Value;
+
+public final class LockStore {
+    private static final String ROW_NAME = "DeletionLock";
+    private final KeyValueService keyValueService;
+
+    public static final LockEntry LOCK_OPEN = ImmutableLockEntry.builder()
+            .rowName(ROW_NAME)
+            .lockId("-1")
+            .reason("Available")
+            .build();
+
+    private LockStore(KeyValueService kvs) {
+        this.keyValueService = kvs;
+    }
+
+    public static LockStore create(KeyValueService kvs) {
```

Comment:
I believe this can cause a few errors on first boot up with people starting up multiple atlas using services in parallel and them concurrently trying to create the table and CAS a lock (whereupon some of them may fail the CAS call because of varying states of old-value)

(It looks like from a quick run-through of your CAS impl in your other commit that making a CASRequest without a set oldValue as is the case here, at least in your C* KVS.cas impl means "oldValue must be empty/non-existant" instead of "oldValue is a don't-care-term / X value")
aka I think the behavior is "succeed if I personally got this lock" instead of "succeed if anyone managed to make me end in a state where the lock is set", but I could be wrong, I didn't thoroughly examine the other PR.